Transport/BacnetIpUdpProtocolTransport.cs

```diff
@@ -158,12 +158,19 @@ private void Open()
         /// </remarks>
         private static void DisableConnReset(UdpClient client)
         {
-            const uint IOC_IN = 0x80000000;
-            const uint IOC_VENDOR = 0x18000000;
-            const uint SIO_UDP_CONNRESET = IOC_IN | IOC_VENDOR | 12;
-
-            client?.Client.IOControl(unchecked((int)SIO_UDP_CONNRESET),
-                new[] { System.Convert.ToByte(false) }, null);
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                const uint IOC_IN = 0x80000000;
+                const uint IOC_VENDOR = 0x18000000;
+                const uint SIO_UDP_CONNRESET = IOC_IN | IOC_VENDOR | 12;
+
+                client?.Client.IOControl(unchecked((int)SIO_UDP_CONNRESET),
+                    new[] { System.Convert.ToByte(false) }, null);
+            }
+            else
+            {
+                client.Client.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
```

Comment:
Are you sure this is the equivalent for other platforms? This CONNRESET setting was needed for windows to prevent error mentioned in the link in comments, but maybe it's not needed on other platforms. Unless you are sure this is needed, I would simply skip doing anything for platforms other than Windows

Reply:
Hi gralin. I cannot be 100% sure that keepalive is actually required in all nonwindows platforms. But it has been working well for ~year now in couple of prod machines running Ubuntu Server 1804. I just translated the functionality here from win -> linux because the original implementation does not work at all (throws when client is created).

Reply:
Ok, of course we nee to fix this for Linux, I'm just wondering if we should blindly set this option or maybe not set anything? and fix it once someone can confirm this is needed?

Reply:
Do you have any exact test I can run that can confirm whether this is needed or not?

Reply:
This problem occured on Windows in the past when application using this library was running for extended period of time and at some point sending or receving anything (don't remember exactly) was throwing exceptions. After research I found this workaround but I don't know if the same problem exists on Linux which has a different implementation of network stack. I would say, if everything works fine on Linux when you comment out this workaround, I would suggest not setting anything unless someone reports a problem.

Reply:
I tested this for ~24h without the fix in linux, seems to work normally

Reply:
Thanks, this version is up on NuGet as beta